src/track/track.cpp

```diff
@@ -27,9 +27,25 @@ std::atomic<int> s_numberOfInstances;
 template<typename T>
 inline bool compareAndSet(T* pField, const T& value) {
     if (*pField != value) {
+        // Copy the value into its final location
         *pField = value;
         return true;
     } else {
+        // Ignore the value if unmodified
+        return false;
+    }
+}
+
+// Overload with a forwarding reference argument for efficiently
+// passing large, movable values.
+template<typename T>
+inline bool compareAndSet(T* pField, T&& value) {
```

Comment:
T&& is here a universal reference. It can be a lvalue or rvalue reference. So the T& version of this function is redundant.

Reply:
:+1: After switching from std::move to std::forward this should work now. Didn't notice that the const-ref version became obsolete.

Reply:
Nope, that doesn't work. The const-ref overload is still needed.

Reply:
Ah OK, probably because of const.